sc2/bot_ai.py

```diff
@@ -306,13 +331,26 @@ def can_afford(self, item_id: Union[UnitTypeId, UpgradeId, AbilityId], check_sup
             cast_range = self._game_data.abilities[ability_id.value]._proto.cast_range
             ability_target = self._game_data.abilities[ability_id.value]._proto.target
             # Check if target is in range (or is a self cast like stimpack)
-            if ability_target == 1 or ability_target == Target.PointOrNone.value and isinstance(target, (Point2, Point3)) and unit.distance_to(target) <= cast_range: # cant replace 1 with "Target.None.value" because ".None" doesnt seem to be a valid enum name
+            if (
+                ability_target == 1
+                or ability_target == Target.PointOrNone.value
+                and isinstance(target, (Point2, Point3))
+                and unit._distance_squared(target.position) <= cast_range ** 2
+            ):  # cant replace 1 with "Target.None.value" because ".None" doesnt seem to be a valid enum name
                 return True
             # Check if able to use ability on a unit
-            elif ability_target in {Target.Unit.value, Target.PointOrUnit.value} and isinstance(target, Unit) and unit.distance_to(target) <= cast_range:
+            elif (
+                ability_target in {Target.Unit.value, Target.PointOrUnit.value}
+                and isinstance(target, Unit)
+                and unit._distance_squared(target.position) <= cast_range ** 2
+            ):
                 return True
             # Check if able to use ability on a position
-            elif ability_target in {Target.Point.value, Target.PointOrUnit.value} and isinstance(target, (Point2, Point3)) and unit.distance_to(target) <= cast_range:
+            elif (
+                ability_target in {Target.Point.value, Target.PointOrUnit.value}
+                and isinstance(target, (Point2, Point3))
+                and unit._distance_squared(target) <= cast_range ** 2
```

Comment:
I get an exception here:

```
Traceback (most recent call last):
  File "/Users/jthomas/git/lambdanaut-sc2/lib/sc2/main.py", line 144, in _play_game_ai
    await ai.on_step(iteration)
  File "/Users/jthomas/git/lambdanaut-sc2/lambdanaut/bot.py", line 121, in on_step
    await self.micro_manager.run()
  File "/Users/jthomas/git/lambdanaut-sc2/lambdanaut/managers/micro.py", line 480, in run
    await self.manage_ravagers()
  File "/Users/jthomas/git/lambdanaut-sc2/lambdanaut/managers/micro.py", line 203, in manage_ravagers
    cached_abilities_of_unit=abilities)
  File "/Users/jthomas/git/lambdanaut-sc2/lib/sc2/bot_ai.py", line 352, in can_cast
    and unit._distance_squared(target) <= cast_range ** 2
AttributeError: 'Unit' object has no attribute '_distance_squared'
```

Reply:
ok i know what to do

Reply:
done

Reply:
Fixed in https://github.com/Dentosal/python-sc2/pull/233/commits/1245c064388c45f49089a450c6503e6c14345f7c